src/plugins/intel_cpu/src/transformations/snippets/x64/pass/set_brgemm_cpu_blocking_params.cpp

```diff
@@ -0,0 +1,110 @@
+        in_desc->set_shape(in_shape);
+    }
+}
+
+void change_out_desc_shape(const std::shared_ptr<ov::Node>& node, const size_t out_idx) {
+    const auto out_desc = PortDescriptorUtils::get_port_descriptor_ptr(node->output(out_idx));
+    const auto& out_shape = node->get_output_shape(out_idx);
+    if (out_desc->get_shape() != out_shape) {
+        out_desc->set_shape(out_shape);
+    }
+}
+} // namespace
+
+pass::SetBrgemmCPUBlockingParams::SetBrgemmCPUBlockingParams() {
+    MATCHER_SCOPE(SetBrgemmCPUBlockingParams);
+
+    auto m_brgemm = ov::pass::pattern::wrap_type<BrgemmCPU>();
+
+    auto callback = [=](ov::pass::pattern::Matcher& m) {
+        OV_ITT_SCOPED_TASK(ov::pass::itt::domains::SnippetsTransform, "ov::intel_cpu::pass::SetBrgemmCPUBlockingParams")
+        const auto node = m.get_match_root();
+        auto brgemm = ov::as_type_ptr<BrgemmCPU>(node);
+        if (brgemm->is_dynamic()) {
+            return false;
+        }
+
+        const auto dimsMatMulIn0 = snippets::utils::get_port_planar_shape(brgemm->input_value(0)).get_shape();
+        const auto dimsMatMulIn1 = snippets::utils::get_port_planar_shape(brgemm->input_value(1)).get_shape();
+        const auto K = *dimsMatMulIn0.rbegin();
+        const auto N = *dimsMatMulIn1.rbegin();
+
+        const auto& input_1_precision = brgemm->get_input_element_type(1);
+
+        // Ticket: 113745
+        // TODO: extend block size selection heuristics
+        const size_t brgemm_block_size_m = 32;
+        const size_t brgemm_block_size_k = [&]() {
+            if (input_1_precision != ov::element::f32)
+                return K;
+            return K > 1024 ? 1024 : K > 512 ? 512 : K;
+        }();
+        const size_t brgemm_block_size_n = input_1_precision != ov::element::f32 ? N : 64;
+
+        brgemm->set_m_block_size(brgemm_block_size_m);
+        brgemm->set_k_block_size(brgemm_block_size_k);
+        brgemm->set_n_block_size(brgemm_block_size_n);
+
+        if (brgemm->is_with_data_repacking()) {
+            const auto brgemm_copy_b = brgemm->get_brgemm_copy();
+
+            const bool isAMXSupported = dnnl::impl::cpu::x64::mayiuse(dnnl::impl::cpu::x64::avx512_core_amx);
+            const auto precision = brgemm_copy_b->get_src_element_type();
+            const auto brgemmVNNIFactor = 4 / precision.size();
+            const bool use_amx = isAMXSupported && precision != ov::element::f32 && (K % brgemmVNNIFactor == 0) && (N % brgemmVNNIFactor == 0);
+
+            const size_t copy_b_block_size_k = use_amx ? brgemm_block_size_k : K;
+            const size_t copy_b_block_size_n = 64;
+
+            brgemm_copy_b->set_k_block_size(copy_b_block_size_k);
+            brgemm_copy_b->set_n_block_size(copy_b_block_size_n);
+            // since N block size affects output shapes, the validation must be called explicitly right after the block size changing
+            brgemm_copy_b->validate_and_infer_types();
+            change_out_desc_shape(brgemm_copy_b, 0);
```

Comment:
BTW Is it possible to update shapes in descriptors inside `validate_and_infer_types()`?

Reply:
I think, it's possible, but does it mean that we must always update descriptors at the shape_inference stage for all nodes inside body? Or `BrgemmCPU`/`BrgemmCopyB` nodes are exceptions and we should update the descriptors at the validation stage only for them?
I made a decision to update the descriptors inside the transformation because this is the single transformation which can change shapes.
We should probably discuss this topic in details offline.

Reply:
Discussed. It was decided to leave the descriptors update logic inside the transformation